packages/strapi-admin/admin/src/containers/Webhooks/EditView/tests/reducer.test.js

```diff
@@ -0,0 +1,265 @@
+  const header = { key: '', value: '' };
+
+  const initialWebhook = {
+    events: [],
+    headers: [header],
+    name: '',
+    url: '',
+  };
+
+  const initialState = fromJS({
+    formErrors: {},
+    initialData: initialWebhook,
+    isTriggering: false,
+    modifiedData: initialWebhook,
+    triggerResponse: {},
+  });
+
+  describe('Format headers object', () => {
+    it('should convert headers object to an array of an empty object if it is empty', () => {
+      const state = initialState;
+
+      const receivedData = {
+        id: 1,
+        name: 'webhook 1',
+        url: 'http://localhost:5000',
+        headers: {},
+        events: ['entry.create', 'entry.update', 'entry.delete'],
+        isEnabled: true,
+      };
+
+      const action = {
+        type: 'GET_DATA_SUCCEEDED',
+        data: receivedData,
+      };
+
+      const data = { ...receivedData, headers: [header] };
+      const expectedState = state
+        .set('initialData', fromJS(data))
+        .set('modifiedData', fromJS(data));
+
+      expect(reducer(state, action)).toEqual(expectedState);
+    });
+
+    it('should convert headers object to an array of objects if it is not empty', () => {
+      const state = initialState;
+
+      const receivedData = {
+        id: 1,
+        name: 'webhook 1',
+        url: 'http://localhost:5000',
+        headers: {
+          accept: 'text/html',
+          authorization: 'Basic YWxhZGRpbjpvcGVuc2VzYW1l',
+        },
+        events: ['entry.create', 'entry.update', 'entry.delete'],
+        isEnabled: true,
+      };
+
+      const action = {
+        type: 'GET_DATA_SUCCEEDED',
+        data: receivedData,
+      };
+
+      const formattedHeaders = [
+        { key: 'accept', value: 'text/html' },
+        { key: 'authorization', value: 'Basic YWxhZGRpbjpvcGVuc2VzYW1l' },
+      ];
+
+      const data = { ...receivedData, headers: formattedHeaders };
+
+      const expectedState = state
+        .update('initialData', () => fromJS(data))
+        .update('modifiedData', () => fromJS(data));
+
+      expect(reducer(state, action)).toEqual(expectedState);
+    });
+  });
+
+  describe('Update Webhook', () => {
+    it('should update modifiedData properly on change', () => {
+      const state = initialState;
+
+      const action = {
+        type: 'ON_CHANGE',
+        keys: ['name'],
+        value: 'new webhook name',
+      };
+
+      const expectedState = state.setIn(
+        ['modifiedData', ...action.keys],
+        action.value
+      );
+
+      expect(reducer(state, action)).toEqual(expectedState);
+    });
+
+    it('should add header to modifiedData correctly', () => {
+      const state = initialState;
+
+      const action = {
+        type: 'ADD_NEW_HEADER',
+        keys: ['headers'],
+      };
+
+      const updatedHeaders = state
+        .getIn(['modifiedData', 'headers'])
+        .push(fromJS(header));
+
+      const expectedState = state.setIn(
```

Comment:
```
const expectedState = state.setIn(['modifiedData', 'headers'], fromJS([{ key: '', value: ''}, {key: '', value: ''}])
```

Doing so will let us be sure that your test is correct and in the case of failure that the code itself doesn't work as expected.